File: programmers/lv.1/완주하지못한선수.js
function solution(participant, completion) {
    var answer = "";
    let map = {};

    for (let i = 0; i < completion.length; i++) {
        if (map[completion[i]]) map[completion[i]]++;
        else map[completion[i]] = 1;
    }

    for (let i = 0; i < participant.length; i++) {
        if (map[participant[i]]) map[participant[i]]--;
        else {
            answer = participant[i];
        }
    }
    return answer;
}

File: programmers/lv.1/완주하지못한선수.java
import java.util.*;
class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        Map <String,Integer> map = new HashMap<>();
        for(String complete : completion){
            if(map.get(complete) == null){
                map.put(complete,1);
            }
            else{
                map.put(complete,map.get(complete)+1);
            }
        }
        for(String participate : participant){
            //System.out.println(participate);
            if(map.get(participate) == null || map.get(participate) == 0  ){
                answer = participate;
            }
            else{
                
                map.put(participate,map.get(participate)-1);
            }
        }
        //System.out.println(map)
        return answer;
    }
}